### PR TITLE
chore(argo-rollouts): Change controller deployment strategy to rollinUpdate

### PR DESCRIPTION
**Pull request by ilyastoliar**
Signed-off-by: Ilya Stoliar <ilyast@monday.com>

Change controller deployment strategy to rollingUpdate due to chore: change controller's deploy strategy to RollingUpdate due to leader election #3334
https://github.com/argoproj/argo-rollouts/pull/3334/

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Review by navele — APPROVED** [review]

**Review by alonbehaim — APPROVED** [review]

**Review by shaharfreidenberg — APPROVED** [review]

**Comment by jmeridth** [comment]
Thank you for your contribution.  Unfortunately, https://github.com/argoproj/argo-rollouts/pull/3334 is merged but not released, so we can't apply it to the helm chart yet.  There was a release after it was merged but it did not include the change.

Also https://github.com/argoproj/argo-helm/pull/2412 is the more configurable approach.

**Review by jmeridth — CHANGES_REQUESTED** [review]
Thank you for your contribution but I believe #2412 is the better approach and we're still waiting for upstream to release the change also.

**Comment by yu-croco** [comment]
Hi @ilyastoliar , thank you for your PR. I agree with @jmeridth .

**Comment by mkilchhofer** [comment]
Closing this in favor of https://github.com/argoproj/argo-helm/pull/2412

**Pull request closed without merging**
